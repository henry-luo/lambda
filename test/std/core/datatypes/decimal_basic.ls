// Test: Decimal Basic Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// Basic arithmetic
let a = 10.5n
a
let b = 2.5n
b
let _sum = a + b
_sum
let diff = a - b
diff
let product = a * b
product
let quotient = a / b
quotient
let remainder = a % b
remainder
let neg = -a
neg
let pos = +a
pos
let gt = a > b
gt
let lt = a < b
lt
let gte = a >= b
gte
let lte = a <= b
lte
let eq = a == b
eq
let neq = a != b
neq
let zero = -0.0n
zero
let one = 1.0n
one
let large = 9999999999999999999999999999.9999999999n
large
let small = 0.0000000000000000000000000001n
small
