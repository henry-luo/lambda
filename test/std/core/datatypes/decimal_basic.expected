10.5
2.5
13.0
8.0
26.25
4.2
0.5
error
10.5
true
false
true
false
false
true
-0.0
1.0
9999999999999999999999999999.9999999999
1E-28
